shared/AppInsightsCore/src/JavaScriptSDK/StatsBeat.ts

```diff
@@ -0,0 +1,182 @@
+                _networkCounter = createNetworkStatsbeat(endpoint);
+                _isEnabled = true;
+                _sdkVersion = version;
+                _getCustomProperties(ikey);
+            }
+
+            _self.isInitialized = (): boolean => {
+                return !!_isEnabled;
+            }
+
+            _self.count = (status: number, payloadData: IPayloadData, endpoint: string) => {
+                if (!_isEnabled || !_checkEndpoint(endpoint)) {
+                    return;
+                }
+                if (payloadData && payloadData["statsBeatData"] && payloadData["statsBeatData"]["startTime"]) {
+                    _networkCounter.totalRequest++;
+                    _networkCounter.requestDuration += utcNow() - payloadData["statsBeatData"]["startTime"];
+                }
+                let retryArray = [401, 403, 408, 429, 500, 503];
+                let throttleArray = [402, 439];
+                if (status === 200) {
+                    _networkCounter.success++;
+                } else if (retryArray.includes(status)) {
+                    _networkCounter.retry[status] = (_networkCounter.retry[status] || 0) + 1;
+                } else if (throttleArray.includes(status)) {
+                    _networkCounter.throttle++;
+                } else if (status !== 307 && status !== 308) {
+                    _networkCounter.failure[status] = (_networkCounter.failure[status] || 0) + 1;
+                }
+                _setupTimer();
+            };
+
+            
+            
+            _self.countException = (endpoint: string, exceptionType: string) => {
+                if (!_isEnabled || !_checkEndpoint(endpoint)) {
+                    return;
+                }
+                _networkCounter.exception[exceptionType] = (_networkCounter.exception[exceptionType] || 0) + 1;
+                _setupTimer();
+            }
+
+            function _setupTimer() {
+                if (!_timeoutHandle) {
+                    _timeoutHandle = scheduleTimeout(() => {
+                        _timeoutHandle = null;
+                        this.trackShortIntervalStatsbeats();
+                    }, STATS_COLLECTION_SHORT_INTERVAL);
+                }
+            }
+
+            _self.trackShortIntervalStatsbeats = (): void => {
+                _trackSendRequestDuration();
+                _trackSendRequestsCount();
+                _networkCounter = createNetworkStatsbeat(_networkCounter.host);
+                _timeoutHandle && _timeoutHandle.cancel();
+                _timeoutHandle = null;
+            }
+
+            function _checkEndpoint(endpoint: string) {
+                return _networkCounter && _networkCounter.host === endpoint;
+            }
+
+            function _getCustomProperties(ikey: string) {
+                _cikey = ikey;
+                _language = STATSBEAT_LANGUAGE;
+                _os = STATSBEAT_TYPE;
+                _runTimeVersion = STATSBEAT_TYPE;
```

Comment:
Is there a more useful value we can track for runtime version for web?

Reply:
Might be able to parse the userAgent to get this information. This might be good default information though if none can be retrieved from that header.

Reply:
As this component is in the core we don't really have access to the existing code that parses out the user-agent, we do a bunch of stuff in the 1DS code and some additional in the OsPlugin.

We could look at enhancing this in a later release -- probably as part of adding the OsPlugin by default (which we are planning in a future release)